src/main/java/io/kabanero/api/InstanceEndpoints.java

```diff
@@ -125,10 +128,50 @@ public Response isAdmin(@PathParam("instanceName") String instanceName) throws I
         return Response.ok(body).build();
     }
 
+    @GET
+    @Path("{instanceName}/admin/list")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAdminLis(@PathParam("instanceName") String instanceName) throws IOException, ApiException, GeneralSecurityException {
+        UserProfile userProfile = UserProfileManager.getUserProfile();
+        String token = userProfile.getAccessToken();
+        GitHubClient client = new GitHubClient();
+        client.setOAuth2Token(token);
+
+        Kabanero instance = KabaneroClient.getAnInstance(instanceName);
+        if (instance == null) {
+            return Response.status(404).entity(new ResponseMessage(instanceName + " not found")).build();
+        }
+
+        String instanceGithubOrg = instance.getSpec().getGithub().getOrganization();
+        TeamService teamService = new TeamService(client);
+
+        Map<Team, List<User>> map = new HashMap<Team, List<User>>();
+        List<User> instanceAdmins = new ArrayList<>();
+        Map<String, Object> body = new HashMap<>();
+
+        for (Team orgTeam : teamService.getTeams(instanceGithubOrg)) {
+            List<User> orgTeamMembers = new ArrayList<>();
+
+            for (User instanceTeamMember : teamService.getMembers(orgTeam.getId())) {
+                orgTeamMembers.add(new UserService(client).getUser(instanceTeamMember.getLogin()));
+                map.put(orgTeam, orgTeamMembers);
+
+                if (!instanceAdmins.stream().filter(member -> instanceTeamMember.getLogin().contains(member.getLogin())).findAny().isPresent()) {
```

Comment:
this seems very verbose, maybe too js like lol. See if you can simplify this

Also using `contains` is not a good idea, if there's an admin `abc` and another user `abc123` then `abc123` might be considered an admin because it contains `abc`

Reply:
also it _seems_ like you're just adding everyone in every team to instanceAdmins (except for duplicates)